src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -164,11 +164,8 @@ public override bool Execute()
         internal /* for testing */ static bool ShouldMergeAnalysisSettings(string language, AnalysisConfig config,
             SonarScanner.MSBuild.Common.ILogger logger)
         {
-            if (string.IsNullOrEmpty(language))
-            {
-                logger.LogInfo(Resources.AnalyzerSettings_LanguageNotSpecified);
-                return false;
-            }
+            Debug.Assert(!string.IsNullOrEmpty(language));
+            Debug.Assert(config != null);
```

Comment:
The refactored code means if either value is null we won't get this far.
I changed the null checks to assertions and removed the redundant test.